pydantic/_internal/_generate_schema.py

```diff
@@ -1905,27 +1949,58 @@ def _call_schema(self, function: ValidateCallSupportedTypes) -> core_schema.Call
 
         TODO support functional validators once we support them in Config
         """
-        sig = signature(function)
-        globalns, localns = self._types_namespace
-        type_hints = _typing_extra.get_function_type_hints(function, globalns=globalns, localns=localns)
+        arguments_schema = self._arguments_schema(function)
+
+        return_schema: core_schema.CoreSchema | None = None
+        config_wrapper = self._config_wrapper
+        if config_wrapper.validate_return:
+            sig = signature(function)
+            return_hint = sig.return_annotation
+            if return_hint is not sig.empty:
+                globalns, localns = self._types_namespace
+                type_hints = _typing_extra.get_function_type_hints(
+                    function, globalns=globalns, localns=localns, include_keys={'return'}
+                )
+                return_schema = self.generate_schema(type_hints['return'])
 
+        return core_schema.call_schema(
+            arguments_schema,
+            function,
+            return_schema=return_schema,
+        )
+
+    def _arguments_schema(
+        self, function: ValidateCallSupportedTypes, parameters_callback: Callable[[int, str, Any], Any] | None = None
+    ) -> core_schema.ArgumentsSchema:
+        """Generate schema for a Signature."""
         mode_lookup: dict[_ParameterKind, Literal['positional_only', 'positional_or_keyword', 'keyword_only']] = {
             Parameter.POSITIONAL_ONLY: 'positional_only',
             Parameter.POSITIONAL_OR_KEYWORD: 'positional_or_keyword',
             Parameter.KEYWORD_ONLY: 'keyword_only',
         }
 
+        sig = signature(function)
+        globalns, localns = self._types_namespace
+        type_hints = _typing_extra.get_function_type_hints(function, globalns=globalns, localns=localns)
+
         arguments_list: list[core_schema.ArgumentsParameter] = []
         var_args_schema: core_schema.CoreSchema | None = None
         var_kwargs_schema: core_schema.CoreSchema | None = None
         var_kwargs_mode: core_schema.VarKwargsMode | None = None
 
-        for name, p in sig.parameters.items():
+        for i, (name, p) in enumerate(sig.parameters.items()):
             if p.annotation is sig.empty:
                 annotation = typing.cast(Any, Any)
             else:
                 annotation = type_hints[name]
 
+            if parameters_callback is not None:
+                from pydantic.validate_call_decorator import SKIP
```

Comment:
```suggestion
                from ..validate_call_decorator import SKIP
```
looks like we mostly do relative imports in this file.

Also shouldn't this be outside the for loop?

Reply:
I wanted to avoid the import unless necessary (if a callback is defined). But I don't mind either way.